pages/docs/concepts/message.md

```diff
@@ -0,0 +1,25 @@
+---
+title: Message
+weight: 10
+---
+
+## What is a message?
+A `message` is a method of communication used to transmit or exchange information from a sender to the receiver through `channels`. A single `message` can be consumed by multiple independent receivers and can also be defined as an _event_ or _command_. The sender includes a payload of data that needs to be processed and serialized by the receiver into an appropriate format, e.g., JSON, XML, binary, etc.
+
+
+``` mermaid
+graph LR
+subgraph one[channel]
+b([message])
+end
+
+A{{sender application}} --> b --> C{{receiver application}}
+  
+```
+
+In the diagram above, the sender application transmits a `message` to the receiver application.
+
+## Messages vs Events
+A `message` carries information from one application to the other, while an `event` is a message that provides details of something that has already occurred. `Messages` contain all necessary information required to perform the request, while `events` only contain data of the changed state.
```

Comment:
There is a contradiction in this paragraph. If an event is a message and an event is only about a changed state, then we can't assure that messages contain all the necessary information to perform a request. Event messages don't contain any information to perform a request.

`Messages` is just a generic term. They can be `events`, `queries`, or `commands`. And `commands` and `queries` are `requests`. 

Here's a representation of what I mean:

```mermaid
graph TD
    A(Message)
    A --> B(Event)
    A --> C(Request)
    C --> D(Query)
    C --> E(Command)
```

Reply:
So how best can we differentiate messages from events?

Reply:
@fmvilas If understand correctly, messages don't necessarily contain information to make a request and events provide information about a changed state. 
So how best can we define/differentiate the two apart from messages being an umbrella term.

Reply:
You can just say that messages contain information but can't anticipate which kind of information. Actually, a message could just contain "Hello Thulie", and that would NOT classify it in any of the described groups (event, query, command). That said, it's not usually the case so we can just simplify and say that, depending on the kind of information the message contains, it may be an event (something that already happened), a query, or a command.

Queries: a query is a request that always expects a response with the results. A query will never ever modify anything on the receiver side. I mean, it's always read-only. It's meant to gather information but never modify it. Example: "give me a list of the last 10 registered users." Notice it doesn't modify anything, just grabs the information and reply.

Command: a command is a request that may or may not receive a response. A command is meant to modify something on the receiver side or trigger some action. Example: "change user email to some@user.com." Notice how it's commanding the receiver to modify data. Depending on the nature of the app, it may or may not reply with the new user information.

Hope that helps :)